example/service.twirp.go

```diff
@@ -678,6 +708,12 @@ func doProtobufRequest(ctx context.Context, client HTTPClient, url string, in, o
 	if err != nil {
 		return wrapInternal(err, "could not build request")
 	}
+	ctx, err = callClientRequestPrepared(ctx, hooks, req)
+	if err != nil {
+		return wrapInternal(err, "failed to call RequestPrepared hook")
```

Comment:
This error text is a little misleading. It isn't necessarily that we failed to call `RequestPrepared` - it's that it returned an error. That's probably because the hook errored - the programmer _wanted_ the error.

Wrapping the error seems a little odd here, too. I'd expect that the error value returned by RequestPrepared is exactly what I'd received in an Error hook, not a wrapped version of it. I'm interested in alternatives to what you have here (or thoughts on whether the API seems correct as-is).

Reply:
@spenczar That's a solid point, I forgot that we're going to make the error into an `InternalError` from the caller anyways, so this is unnecessary. I'll remove it and just return the `err` itself.